### PR TITLE
Bump Go to 1.18

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
golangci-lint@1.45 now supports go1.18 we are good to review this
/hold  cancel

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/jaypipes/ghw v0.8.1-0.20210827132705-c7224150a17e
 	github.com/k8stopologyawareschedwg/noderesourcetopology-api v0.0.12
 	github.com/klauspost/cpuid/v2 v2.0.11
-	github.com/onsi/ginkgo v1.14.0
+	github.com/onsi/ginkgo v1.16.5
```
Comment (marquiz) [inline go.mod:14]: Updating the dependencies would be better to do in a separate PR. Also, let's keep these in sync with k/k

Comment (ArangoGutierrez) [reply go.mod]: done

**Comment by ArangoGutierrez** [comment]
Test-infra pair -> https://github.com/kubernetes/test-infra/pull/25651

**Comment by ArangoGutierrez** [comment]
@marquiz can this make it to v0.11? or defer?

**Comment by marquiz** [comment]
> @marquiz can this make it to v0.11? or defer?

I suggest to defer to minimize risks and hassle on that front

**Comment by ArangoGutierrez** [comment]
> > @marquiz can this make it to v0.11? or defer?
> 
> I suggest to defer to minimize risks and hassle on that front

v0.11 is cut, we can circle back to this

File: Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ ARG BASE_IMAGE_FULL
 ARG BASE_IMAGE_MINIMAL
 
 # Build node feature discovery
-FROM golang:1.17.2-buster as builder
+FROM golang:1.18-buster as builder
```
Comment (marquiz) [inline Dockerfile:5]: nit: let's continue to use a pinpointed version for reproducible builds

```suggestion
FROM golang:1.18.0-buster as builder
```

**Review by marquiz — APPROVED** [review]
Thanks @ArangoGutierrez, looks good 👍

Let's wait for https://github.com/kubernetes/test-infra/pull/25651 to land first.
/hold

**Comment by kad** [comment]
You need to make sure that 1.18.1+ is used. 1.18.0 had regression in PEM certificates handling which was fixed very recently.

**Comment by marquiz** [comment]
> You need to make sure that 1.18.1+ is used. 1.18.0 had regression in PEM certificates handling which was fixed very recently.

Yeah. Looks like the docker images were also published a few hours ago

**Pull request merged**